pkg/sensors/tracing/kprobe_test.go

```diff
@@ -5095,3 +5097,123 @@ spec:
 		t.Fatalf("GetDefaultObserverWithFile error: %s", err)
 	}
 }
+
+// Test module loading/unloading on Ubuntu
+func TestTraceKernelModule(t *testing.T) {
+	var doneWG, readyWG sync.WaitGroup
+	defer doneWG.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
+	defer cancel()
+
+	hookFull := `apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "monitor-kernel-modules"
+spec:
+  kprobes:
+  - call: "security_kernel_read_file"
+    # Explicit module loading using file descriptor finit_module() to print module full path
+    syscall: false
+    return: true
+    args:
+    - index: 0
+      type: "file"
+    - index: 1
+      type: "int"
+    returnArg:
+      index: 0
+      type: "int"
+    selectors:
+    - matchArgs:
+      - index: 1
+        operator: "Equal"
+        values:
+        - "2"  # READING_MODULE
+  - call: "do_init_module"
+    syscall: false
+    args:
+    - index: 0
+      type: "module"
+  - call: "free_module"
+    syscall: false
+    args:
+    - index: 0
+      type: "module"
+`
+
+	// This test works only on Ubuntu
```

Comment:
This means that it will run on `Tetragon Go Test` but not on `vmtest` in our CI. I believe that this is enough for now. But, I am just curious on what is the reason that this does not run on `vmtests`. I believe that we can install any packages that we may miss, or choose a different module in the case where xfs is not available.

Reply:
> This means that it will run on `Tetragon Go Test` but not on `vmtest` in our CI. I believe that this is enough for now. But, I am just curious on what is the reason that this does not run on `vmtests`. I believe that we can install any packages that we may miss, or choose a different module in the case where xfs is not available.

Indeed and it is fairly enough as a first setup as we cover what we need.

Checking if modules are signed, will need to build our kernels with right configs then build the module and sign it which is automatically handled by some kernel parameters but it needs more work. Ubuntu distro already ships signed modules so we are covered.

I added the task to build a local kernel module and test it as next https://github.com/cilium/tetragon/issues/1433 on all platform.

But yeh we are covered for now. Thanks for the review ;-)

Reply:
I'm not sure if we have package manager on vmtests, but also we don't build xfs , and I did chose that one for the kubernetes documentation as it is easy on pods...